src/collection.js

```diff
@@ -600,6 +612,16 @@ export default class Collection {
       lastModified: options.lastModified,
       headers: options.headers
     })
+      // Check if server is empty (flushed or changed).
+      .then(changes => {
+        const firstSync =           (!this.lastModified);
+        const lastModifiedUpdated = (changes.lastModified !== this.lastModified);
+        const emptyCollection =     (changes.changes.length === 0);
```

Comment:
From what I read, it seems we assume a collection is empty if no changes are retrieved from the server; might be worth a comment to avoid confusion.